### PR TITLE
Add initial support for Github Apps.

### PR DESCRIPTION
**Pull request by wrecker**
Needed for #1621

### CODE REVIEW

**Comment by sfdye** [comment]
It seems there is an isort issue, you can get around by `tox -e lint` locally

**Comment by wrecker** [comment]
Ahh.. I can't `tox -e lint` because it keeps looking for python3.6. My dev environment is ubuntu20.04 which doesn't have 3.6 in its package repo. I have to look into running 3.6 with docker.

File: github/GithubApp.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+#                                                                              #
+################################################################################
+
+import github.GithubObject
+import github.NamedUser
+
+
+class GithubApp(github.GithubObject.CompletableGithubObject):
+    """
+    This class represents check suites. The reference can be found here https://docs.github.com/en/rest/reference/apps
```
Comment (s-t-e-v-e-n-k) [inline github/GithubApp.py:31]: This class does not represent check suites.

Comment (wrecker) [reply github/GithubApp.py]: :grimacing: Oops!

**Comment by wrecker** [comment]
@s-t-e-v-e-n-k Can I get a OK / NOT OK for this PR. my next set of PRs reference the changes in this one.

**Pull request merged**